### PR TITLE
add mish and softplus

### LINKED ISSUE #246: Feature Request: Add Mish activation
Mish is a new novel activation function proposed in this [paper](https://arxiv.org/abs/1908.08681).
It has shown promising results so far and has been adopted in several packages including:

- [TensorFlow-Addons](https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations)
- [SpaCy (Tok2Vec Layer)](https://github.com/explosion/spaCy)
- [Thinc - SpaCy's official NLP based ML library](https://github.com/explosion/thinc/releases/tag/v7.3.0)
- [Echo AI](https://github.com/digantamisra98/Echo)
- [CNTKX - Extension of Microsoft's CNTK](https://github.com/delzac/cntkx)
- [FastAI-Dev](https://github.com/fastai/fastai_dev/blob/0f613ba3205990c83de9dba0c8798a9eec5452ce/dev/local/layers.py#L441)
- [Darknet](https://github.com/AlexeyAB/darknet/commit/bf8ea4183dc265ac17f7c9d939dc815269f0a213)
- [Yolov3](https://github.com/ultralytics/yolov3/commit/444a9f7099d4ff1aef12783704e3df9a8c3aa4b3)
- [BeeDNN - Library in C++](https://github.com/edeforas/BeeDNN)
- [Gen-EfficientNet-PyTorch](https://github.com/rwightman/gen-efficientnet-pytorch)
- [dnet](https://github.com/umangjpatel/dnet/blob/master/nn/activations.py)

All benchmarks, analysis and links to official package implementations can be found in this [repository](https://github.com/digantamisra98/Mish) 

It would be nice to have Mish as an option within the activation function group.

This is the comparison of Mish with other conventional activation functions in a SEResNet-50 for CIFAR-10: 
se50_1

### PR DESCRIPTION
**Pull request by tscholak**
closes #246 

@digantamisra98 added as requested.

### CODE REVIEW

**Comment by digantamisra98** [comment]
Thanks for considering the request!

File: hasktorch/src/Torch/Typed/Native.hs [hunk 1]
```diff
@@ -311,6 +311,20 @@ selu
...
+-- (Float,([3,2],[[0.86509836,0.86509836],[0.86509836,0.86509836],[0.86509836,0.86509836]]))
+mish
+  :: forall shape dtype device
+   . ( StandardFloatingPointDTypeValidation device dtype
+     , BasicArithmeticDTypeIsValid device dtype
+     , shape ~ Broadcast shape shape
+     )
+  => Tensor device dtype shape
+  -> Tensor device dtype shape
+mish = mul =<< tanh . softplus (1 :: Float) 20
```
Comment (junjihashimoto) [inline hasktorch/src/Torch/Typed/Native.hs]: Both 1 and 20 are just default values of softplus.

Comment (tscholak) [reply hasktorch/src/Torch/Typed/Native.hs]: yes, that's right

Comment (tscholak) [reply hasktorch/src/Torch/Typed/Native.hs]: I don't think @digantamisra98 specified what these should be in his paper. Thus, I went with the default.

Comment (tscholak) [reply hasktorch/src/Torch/Typed/Native.hs]: nvm, he defined `softplus x = ln (1 + e^x)`, so that settles it.

Comment (junjihashimoto) [reply hasktorch/src/Torch/Typed/Native.hs]: Oh, should I revert this merge?

Comment (junjihashimoto) [reply hasktorch/src/Torch/Typed/Native.hs]: I'm sorry to merge early.

Comment (tscholak) [reply hasktorch/src/Torch/Typed/Native.hs]: No, it’s fine the way it is. Thank you!

**Pull request merged**